### PR TITLE
Use system.allowed_oembed config to allow iframe source

### PR DESCRIPTION
**Pull request by MrPetovan**
Replaces #10044 
Address part 1 of https://github.com/friendica/friendica/issues/10019#issuecomment-798931999

### CODE REVIEW

File: src/Content/Text/HTML.php [hunk 1]
```diff
@@ -961,4 +961,61 @@ public static function unamp($s)
...
+		$config->set('HTML.Doctype', 'HTML5');
+
+		// Used to remove iframe with src attribute filtered out
+		$config->set('AutoFormat.RemoveEmpty', true);
+
+		$config->set('HTML.SafeIframe', true);
+
+		array_walk($allowedIframeDomains, function (&$domain) {
+			// Allow the domain and all its eventual sub-domains
+			$domain = 'https://(?:(?!-)[A-Za-z0-9-]{1,63}(?<!-)\.)*' . preg_quote($domain, '%');
```
Comment (annando) [inline src/Content/Text/HTML.php]: Wouldn't this regular expression also be valid when you enter `youtube.com` but the URL points to `https://fakeyoutube.com` or `https://youtube.com.fake.tld`?

**Comment by annando** [comment]
To fix the test errors, you have to enter this here: https://github.com/friendica/friendica/pull/10044/files#diff-29a560f2e02fcf1abb799b8362cc75a8d6a12ce4dd4c07a3d88033ebe9f23cf9R66-R71

File: src/Content/Text/HTML.php (hunk 1 above)
Comment (MrPetovan) [reply src/Content/Text/HTML.php]: The first wouldn't be allowed, but the second would, I have to add a trailing `/` to the expression. Good catch!

**Review by annando — APPROVED** [review]

File: src/Content/Text/HTML.php (hunk 1 above)
Comment (MrPetovan) [reply src/Content/Text/HTML.php]: This has been addressed.

**Comment by MrPetovan** [comment]
Yesss!

**Comment by annando** [comment]
@MrPetovan tests really are fun, aren't they? :grimacing:

**Comment by MrPetovan** [comment]
They make sense, but I forget about them every time.

**Comment by AlfredSK** [comment]
Has the logic of the configuration been changed with this PR? Will YT and Vimeo videos still work with OEmbed disabled? As far as I can see both are somehow hardcoded.

**Comment by MrPetovan** [comment]
The logic has been extended, which means that what was working before will still work. What wasn't working before can now work provided changes to the now-called "Trusted domain list". I kept the Youtube and Vimeo exceptions because we provide specific BBCodes for both.

**Pull request merged**